src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,676 @@
+        playerDeathPopulationPenaltyReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.PlayerDeathPopulationPenalty = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnGlucoseDecayRateValueChanged(double percentage)
+    {
+        percentage = Math.Round(percentage, 2);
+        var percentageFormat = TranslationServer.Translate("PERCENTAGE_VALUE");
+        glucoseDecayRateReadout.Text = string.Format(CultureInfo.CurrentCulture, percentageFormat,
+            percentage);
+        settings.GlucoseDecay = (float)percentage * 0.01f;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnOsmoregulationMultiplierValueChanged(double amount)
+    {
+        amount = Math.Round(amount, 1);
+        osmoregulationMultiplierReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.OsmoregulationMultiplier = (float)amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnFreeGlucoseCloudToggled(bool pressed)
+    {
+        settings.FreeGlucoseCloud = pressed;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnLifeOriginSelected(int index)
+    {
+        // Set both buttons here as we only received a signal from one of them
+        lifeOriginButton.Selected = index;
+        lifeOriginButtonAdvanced.Selected = index;
+
+        settings.Origin = (WorldGenerationSettings.LifeOrigin)index;
+    }
+
+    private void OnMapTypeSelected(int index)
+    {
+        settings.MapType = MapTypeIndexToValue(index);
+    }
+
+    private WorldGenerationSettings.PatchMapType MapTypeIndexToValue(int index)
+    {
+        switch (index)
+        {
+            case 1:
+                return WorldGenerationSettings.PatchMapType.Classic;
+            default:
+                return WorldGenerationSettings.PatchMapType.Procedural;
+        }
+    }
+
+    private void OnLAWKToggled(bool pressed)
+    {
+        // Set both buttons here as we only received a signal from one of them
+        lawkButton.Pressed = pressed;
+        lawkAdvancedButton.Pressed = pressed;
+
+        settings.LAWK = lawkButton.Pressed;
+
+        UpdateLifeOriginOptions(pressed);
+    }
+
+    private void UpdateLifeOriginOptions(bool lawk)
+    {
+        // If we've switched to LAWK only, disable panspermia
+        var panspermiaIndex = (int)WorldGenerationSettings.LifeOrigin.Panspermia;
+        lifeOriginButton.SetItemDisabled(panspermiaIndex, lawk);
+        lifeOriginButtonAdvanced.SetItemDisabled(panspermiaIndex, lawk);
+
+        // If we had selected panspermia, reset to vents
+        if (lawk && lifeOriginButton.Selected == panspermiaIndex)
+        {
+            lifeOriginButton.Selected = (int)WorldGenerationSettings.LifeOrigin.Vent;
+            lifeOriginButtonAdvanced.Selected = (int)WorldGenerationSettings.LifeOrigin.Vent;
+        }
+    }
+
+    private void OnGameSeedChangedFromBasic(string text)
+    {
+        gameSeedAdvanced.Text = text;
```

Comment:
Can't this and OnGameSeedChangedFromAdvanced be combined? Same as most of the other callbacks or is there a reason why this is this way?

Reply:
Ah, after making this change I've realised why I couldn't do that. Since a combined method sets the text of both text inputs, it would in particular overwrite the text the user has just entered. Although it overwrites to the correct value, it places the caret back at the start of the box and won't let you change the caret position. So unless you have a suggestion for that, I'll revert to the way it was.